### DISCUSSION TITLE
@next/bundle-analyzer not opening client/package web pages displaying file sizes

### DISCUSSION THREAD

**Discussion by jamesryan-dev** [vercel/next.js#16884] [Help]
Have follow the docs implementation:

`next.config.js`

```
const withPlugins = require('next-compose-plugins');
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
})


module.exports = withPlugins([
  [withBundleAnalyzer({})],
...
```

and whilst in the terminal I am getting a compiled website and report upon running:

▶ npm run analyze

from 

```
    "analyze": "cross-env ANALYZE=true next build",
    "analyze:server": "cross-env BUNDLE_ANALYZE=server next build",
    "analyze:browser": "cross-env BUNDLE_ANALYZE=browser next build"
```

have tried to two other alternatives but nothing seems to open a webpage in my browser.. 

using chrome if that makes a difference

Any help greatly appreciated..

trying to optimise my site and unsure where after a few rounds i can improve so was hoping this library could resolve this uncertantiy..

**Comment by jamesryan-dev** [comment]
<img width="204" alt="Screenshot 2020-09-06 at 14 27 39" src="https://user-images.githubusercontent.com/13485129/92326859-20b37000-f04d-11ea-8aeb-af47c1c0cae6.png">

upon building via the analyze command my .next doesnt seem to have a analyze dir also.. 🤔

**Reply by MAfzalKhan1997** [reply]
any solution ?

**Reply by jamesryan-dev** [reply]
no sorry - i couldn't find a working solution

**Reply by seanblonien** [reply]
~~I am also running into this problem. I am in a monorepo which may have something to do with it~~

Actually it's working for me now! I used to have this 
`cross-env ANALYZE=true && next build`
but I changed it to
`cross-env ANALYZE=true next build`

_Also_, I did run the build command from the root of my monorepo which may have changed things.

Other than that, here is what my `next.config.js` file looks like, although I don't think these config settings are relevant anyways (maybe webpack 5 is significant):
<details>
<summary>next.config.js</summary>

```js
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
});

const useWebpack5 = true;
const config = {
  reactStrictMode: true,
  poweredByHeader: false,
  future: {
    webpack5: useWebpack5,
  },
  webpack: (config, {isServer}) => {
    if (!isServer) {
      if (useWebpack5) {
        config.resolve.fallback.fs = false;
      } else {
        config.node = {
          fs: 'empty',
        };
      }
    }
    return config;
  },
};
module.exports = withBundleAnalyzer(config);
```

</details>

**Comment by smo043** [comment]
I resolved this issue by adding bundle analyzer plugin at the end

`module.exports = withPlugins([withImages, withTM(), [withBundleAnalyzer]], {});`

**Reply by rsipakov** [reply]
For me the same if using the `withPlugins`.

**Reply by omkar4253** [reply]
Worked for me!

**Reply by juddroid** [reply]
@rsipakov  THX, Worked for me, too!
but why it worked ?

**Comment by arbaouimehdi** [comment]
**Side note:** 
if you’re running `next build --turbopack`, the bundle analyzer won’t open because **Turbopack** doesn’t run **Webpack** or its plugins, and `@next/bundle-analyzer` is a Webpack plugin.

Fix: omit the flag and build with Webpack:
```
# works (Webpack)
ANALYZE=true next build

# won’t open analyzer (Turbopack)
ANALYZE=true next build --turbopack
```

And keep two scripts `build` with `--turbopack` and  `analyze` without it.
